libraries/c_sdk/standard/mqtt/src/iot_mqtt_serializer_deserializer_wrapper.c

```diff
@@ -0,0 +1,772 @@
+#include "iot_config.h"
+
+/* Standard includes. */
+#include <string.h>
+
+/* Error handling include. */
+#include "private/iot_error.h"
+
+/* MQTT internal includes. */
+#include "private/iot_mqtt_internal.h"
+
+/* MQTT v4_beta 2 lightweight library includes. */
+#include "mqtt_lightweight.h"
+
+/* Atomic operations. */
+#include "iot_atomic.h"
+
+/*-----------------------------------------------------------*/
+
+/*Size of Puback packet */
+#define MQTT_PACKET_PUBACK_SIZE      ( 4 )
+
+/**
+ * @brief Per the MQTT 3.1.1 spec, the largest "Remaining Length" of an MQTT
+ * packet is this value.
+ */
+#define MQTT_MAX_REMAINING_LENGTH    ( 268435455UL )
+
+/*-----------------------------------------------------------*/
+
+/*Generate Id for packet*/
+static uint16_t _nextPacketIdentifier( void );
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Calculate the number of bytes required to encode an MQTT
+ * "Remaining length" field.
+ *
+ * @param[in] length The value of the "Remaining length" to encode.
+ *
+ * @return The size of the encoding of length. This is always `1`, `2`, `3`, or `4`.
+ */
+static size_t _remainingLengthEncodedSize( size_t length );
+
+/*-----------------------------------------------------------*/
+
+/*Generate Id for packet*/
+static uint16_t _nextPacketIdentifier( void )
+{
+    /* MQTT specifies 2 bytes for the packet identifier; however, operating on
+     * 32-bit integers is generally faster. */
+    static uint32_t nextPacketIdentifier = 1;
+
+    /* The next packet identifier will be greater by 2. This prevents packet
+     * identifiers from ever being 0, which is not allowed by MQTT 3.1.1. Packet
+     * identifiers will follow the sequence 1,3,5...65535,1,3,5... */
+    return ( uint16_t ) Atomic_Add_u32( &nextPacketIdentifier, 2 );
+}
+
+/*-----------------------------------------------------------*/
+
+static size_t _remainingLengthEncodedSize( size_t length )
+{
+    size_t encodedSize = 0;
+
+    /* length should have already been checked before calling this function. */
+    IotMqtt_Assert( length <= MQTT_MAX_REMAINING_LENGTH );
+
+    /* Determine how many bytes are needed to encode length.
+     * The values below are taken from the MQTT 3.1.1 spec. */
+
+    /* 1 byte is needed to encode lengths between 0 and 127. */
+    if( length < 128 )
+    {
+        encodedSize = 1;
+    }
+    /* 2 bytes are needed to encode lengths between 128 and 16,383. */
+    else if( length < 16384 )
+    {
+        encodedSize = 2;
+    }
+    /* 3 bytes are needed to encode lengths between 16,384 and 2,097,151. */
+    else if( length < 2097152 )
+    {
+        encodedSize = 3;
+    }
+    /* 4 bytes are needed to encode lengths between 2,097,152 and 268,435,455. */
+    else
+    {
+        encodedSize = 4;
+    }
+
+    return encodedSize;
+}
+
+/*-----------------------------------------------------------*/
+
+uint8_t _IotMqtt_GetPacketType( void * pNetworkConnection,
```

Comment:
Since it is a new file, let's keep same coding style for this file. Either make all these functions _IotMqtt_XYZ or xYZ() ( get rid of ' _ '  )

Reply:
Ok, will keep it as _IotMqtt_XYZ